### PR TITLE
[GEARPUMP-226] refactor build files

### PR DESCRIPTION
**Pull request by manuzhang**
Refactored build files as follows

1. break `Build.scala` into parts for different module categories, e.g. `BuildExternals`, `BuildDashboard`.
2. libraries to shade are now added as managed dependencies, shaded with their depending projects and finally assembled to a uber jar. For example, `gearpump-core-$version.jar` will now contain both the gearpump sources and shaded libraries, whose package names are both `org.apache.gearpump`.  The original dependencies are excluded from `pack`. 
3. package names in the codebase are changed as needed by 2
4. `gearpump-core` is now `provided` for `gearpump-streaming` and won't be transitive dependencies for projects depending on`gearpump-streaming`. `gearpump-core` needs to be explicitly included as `provided` dependency so I add it to the published pom files for users.

### CODE REVIEW

**Comment by manuzhang** [comment]
@huafengw please review.

File: integrationtest/core/src/it/scala/org/apache/gearpump/integrationtest/suites/StandaloneModeSuite.scala [hunk 1]
```diff
@@ -28,14 +28,14 @@ import org.apache.gearpump.integrationtest.minicluster.MiniCluster
...
-  new ConnectorKafkaSpec,
-  new MessageDeliverySpec
+  // new CommandLineSpec,
+  // new RestServiceSpec,
+  new ExampleSpec //,
+  // new DynamicDagSpec,
+  // new StormCompatibilitySpec,
+  // new StabilitySpec,
+  // new ConnectorKafkaSpec,
+  // new MessageDeliverySpec
```
Comment (huafengw) [inline integrationtest/core/src/it/scala/org/apache/gearpump/integrationtest/suites/StandaloneModeSuite.scala]: unnecessary change

File: project/BuildExperiments.scala [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      ))
+    .dependsOn(core % "provided", streaming % "test->test; provided")
+
+  lazy val redis = Project(
+    id = "gearpump-experiments-redis",
+    base = file("experiments/redis"),
+    settings = commonSettings ++ noPublish ++
+      Seq(
+        libraryDependencies ++= Seq(
+          "redis.clients" % "jedis" % "2.9.0"
```
Comment (huafengw) [inline project/BuildExperiments.scala]: extract the redis version

File: project/BuildGearpump.scala [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+          case x =>
+            val oldStrategy = (assemblyMergeStrategy in assembly).value
+            oldStrategy(x)
+        },
+
+        libraryDependencies ++= Seq(
+          "com.goldmansachs" % "gs-collections" % gsCollectionsVersion
+        ),
+
+        pomPostProcess := {
```
Comment (huafengw) [inline project/BuildGearpump.scala]: do we need post pom process any more?

Comment (huafengw) [inline project/BuildGearpump.scala]: I don't get it. These shaded library are already been assembled into the gearpump-core jar file right, which means there should not be any dependency any more?

Comment (manuzhang) [inline project/BuildGearpump.scala]: they are now transitive dependencies and will be added the published pom file

File: project/Pack.scala [hunk 4]
```diff
@@ -116,11 +118,21 @@ object Pack extends sbt.Build {
           "lib/storm" -> new ProjectsToPack(storm.id).exclude(streaming.id)
         ),
         packExclude := Seq(thisProjectRef.value.project),
+        packExcludeJars := Seq(
+          "akka-kryo-.*\\.jar",
+          "gs-collections-.*\\.jar",
+          "guava-.*\\.jar",
```
Comment (huafengw) [inline project/Pack.scala]: There should be a guava artifact under `lib/yarn`, now after you exclude this when packing, does it still exist?

**Comment by huafengw** [comment]
+1
